src/middleware/builtin.ts

```diff
@@ -73,31 +76,37 @@ export const onlyEvents: Middleware<AnyMiddlewareArgs & { event?: SlackEvent }>
 /**
  * Middleware that checks for matches given constraints
  */
-export function matchActionConstraints(
+export function matchConstraints(
     constraints: ActionConstraints,
     // TODO: this function signature could be wrong. this gets used in options() so the action property can be undefined
-  ): Middleware<SlackActionMiddlewareArgs<SlackAction>> {
-  return ({ action, next, context }) => {
-    // NOTE: DialogSubmitAction and MessageAction do not have an actions array
-    const innerAction: InteractiveAction | UnknownElementAction<string> | undefined =
-      action.actions !== undefined ? action.actions[0] : undefined;
+): Middleware<AnyMiddlewareArgs> {
+  return (args) => {
+    // NOTE: DialogSubmitAction, MessageAction, and Options requests do not have an actions array
```

Comment:
i think a more useful comment would be something like

```suggestion
    // NOTE: DialogSubmitAction and MessageAction do not have an innerPayload
```